src/Joystick/Joystick.cc

```diff
@@ -198,6 +286,42 @@ void Joystick::_saveSettings(void)
     }
 }
 
+// Relative mappings of axis functions between different TX modes
+int Joystick::_mapFunctionMode(int mode, int function) {
+
+    static const int mapping[][4] = {
+        { 2, 1, 0, 3 },
+        { 2, 3, 0, 1 },
+        { 0, 1, 2, 3 },
+        { 0, 3, 2, 1 }};
+
+    return mapping[mode-1][function];
+}
+
+// Remap current axis functions from current TX mode to new TX mode
+void Joystick::_remapAxes(int currentMode, int newMode, int (&newMapping)[maxFunction]) {
+    int temp[maxFunction];
+
+    for(int function = 0; function < maxFunction; function++) {
+        temp[_mapFunctionMode(newMode, function)] = _rgFunctionAxis[_mapFunctionMode(currentMode, function)];
+    }
+
+    for(int function = 0; function < maxFunction; function++) {
+        newMapping[function] = temp[function];
+    }
+
+}
+
+void Joystick::setTXMode(int mode) {
```

Comment:
I think this reflects a difference in joysticks and RC transmitters, RC transmitters may come built for a specific mode that is hard-wired. You are selecting the mode that corresponds to the RC transmitter in your hands. Joysticks aren't made this way, so you are selecting the mode that you want your joystick to emulate.

If we remove this functionality, then we won't be able to auto-calibrate gamecontrollers, we won't be able to have vehicle-specific settings, and we will have to recalibrate the joystick in order to change the mode. I don't know if it will be clear that you need to recalibrate joysticks in order to change the mode either. We could have a message somewhere that says you must recalibrate the joystick to change the mode. Or we could disable the mode selection at all times except, as the first step in calibration.

In my opinion it is 'good functionality', but I understand your concerns. It is not absolutely necessary to leave this in. I did structure the commits so it would be straightforward to take out or revert.